pkg/util/shell/shell_test.go

```diff
@@ -0,0 +1,217 @@
+/*
+Copyright (C) 2017 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package shell
+
+import (
+	"os"
+	"testing"
+)
+
+type shellTestCase struct {
+	name         string
+	cmdLine      string
+	expectedHint string
+}
+
+type prefixSuffixDelimiterTestCase struct {
+	shellName string
+	prefix    string
+	suffix    string
+	pathVar   bool
+	delimiter string
+}
+
+func TestUnknownShell(t *testing.T) {
+	expectedShellName := "foo"
+	defer func(shell string) { os.Setenv("SHELL", shell) }(os.Getenv("SHELL"))
+	os.Setenv("SHELL", "/bin/"+expectedShellName)
+
+	shell, err := GetShell("")
+	if err != nil {
+		t.Fatal("Unexpected err : ", err.Error())
+	}
+
+	if shell != expectedShellName {
+		t.Fatalf("Expected shell to be %s but %s", expectedShellName, shell)
+	}
+}
+
+func TestKnownShell(t *testing.T) {
+	expectedShellName := "bash"
+	shell, err := GetShell("bash")
+	if err != nil {
+		t.Fatal("Unexpected err : ", err.Error())
+	}
+
+	if shell != expectedShellName {
+		t.Fatalf("Expected shell to be %s but %s", expectedShellName, shell)
+	}
+}
+
+func TestNoProxyFromEnvWhenUsedLowerCase(t *testing.T) {
+	expectedNoProxyVarName := "no_proxy"
+	expectedNoProxyVarValue := "FOO_BAR"
+	defer func(val string) { os.Setenv(expectedNoProxyVarName, val) }(os.Getenv(expectedNoProxyVarName))
+	os.Setenv(expectedNoProxyVarName, expectedNoProxyVarValue)
+
+	_, noProxyValue := FindNoProxyFromEnv()
+	if noProxyValue != expectedNoProxyVarValue {
+		t.Fatalf("Expected no proxy var value as %s but got %s", expectedNoProxyVarValue, noProxyValue)
+	}
+}
+
+func TestNoProxyFromEnvWhenUsedUpperCase(t *testing.T) {
+	expectedNoProxyVarName := "NO_PROXY"
+	expectedNoProxyVarValue := "FOO_BAR"
+	defer func(val string) { os.Setenv(expectedNoProxyVarName, val) }(os.Getenv(expectedNoProxyVarName))
+	os.Setenv(expectedNoProxyVarName, expectedNoProxyVarValue)
+
+	_, noProxyValue := FindNoProxyFromEnv()
+	if noProxyValue != expectedNoProxyVarValue {
+		t.Fatalf("Expected no proxy var value as %s but got %s", expectedNoProxyVarValue, noProxyValue)
+	}
+}
+
+func TestGenerateUsageHint(t *testing.T) {
```

Comment:
> For this, I would like to raise it in new issue and implementation as part of oc_env_test.go and env_test.go

Why? The very issue was from the start that _oc-env_ returned invalid output for _fish_. So I would very much think that a _oc_env_test.go_ is in scope here.